### PR TITLE
done the mvp and a bit of stretch

### PR DESCRIPTION
**Pull request by justman00**
Sorry for not doing more than one commit, The project was small and it went out of my mind

### CODE REVIEW

File: assignments/prototypes.js [hunk 1]
```diff
@@ -7,13 +7,19 @@
...
   Each constructor function has unique properties and methods that are defined in their block comments below:
 */
-  
+
 /*
   === GameObject ===
   * createdAt
   * dimensions (These represent the character's size in the video game)
   * destroy() // prototype method -> returns the string: 'Object was removed from the game.'
 */
```
Comment (GannonDetroit) [inline assignments/prototypes.js:16]: good job with these.

File: assignments/prototypes.js [hunk 2]
```diff
@@ -22,6 +28,15 @@
   * takeDamage() // prototype method -> returns the string '<object name> took damage.'
   * should inherit destroy() from GameObject's prototype
 */
```
Comment (GannonDetroit) [inline assignments/prototypes.js:30]: good job here

File: assignments/prototypes.js [hunk 3]
```diff
@@ -32,79 +47,102 @@
   * should inherit destroy() from GameObject through CharacterStats
   * should inherit takeDamage() from CharacterStats
 */
- 
+
```
Comment (GannonDetroit) [inline assignments/prototypes.js:50]: good job

File: assignments/prototypes.js [hunk 4]
```diff
@@ -32,79 +47,102 @@
...
+    height: 1
+  },
+  healthPoints: 5,
+  name: "Bruce",
+  team: "Mage Guild",
+  weapons: ["Staff of Shamalama"],
+  language: "Common Tongue"
+});
 
-  const swordsman = new Humanoid({
```
Comment (GannonDetroit) [inline assignments/prototypes.js:61]: what extensions are you using in VS code? 

Is there a reason these things are being deleted like this and added back ?

File: assignments/prototypes.js [hunk 5]
```diff
@@ -32,79 +47,102 @@
...
+console.log(swordsman.healthPoints); // 15
+console.log(mage.name); // Bruce
+console.log(swordsman.team); // The Round Table
+console.log(mage.weapons); // Staff of Shamalama
+console.log(archer.language); // Elvish
+console.log(archer.greet()); // Lilith offers a greeting in Elvish.
+console.log(mage.takeDamage()); // Bruce took damage.
+console.log(swordsman.destroy()); // Sir Mustachio was removed from the game.
+
+// Stretch task:
```
Comment (GannonDetroit) [inline assignments/prototypes.js:122]: you did half the stretch, it technically wants you to make a villain AND a hero. But I still respect that you attempted and did some of the stretch. Good job.

File: assignments/this.js [hunk 6]
```diff
@@ -1,26 +1,53 @@
...
-*
-* 1. 
-* 2. 
-* 3. 
-* 4. 
-*
-* write out a code example of each explanation above
-*/
+ * in your own words. explain the four principle for the "this" keyword below.
+ *
```
Comment (GannonDetroit) [inline assignments/this.js:3]: lol, simple and sweet answers. 

Well done.

File: assignments/this.js [hunk 7]
```diff
@@ -1,26 +1,53 @@
...
+ * 4. it is used in constructors
+ *
+ * write out a code example of each explanation above
+ */
 
 // Principle 1
-
 // code example for Window Binding
-
+var x = this;
```
Comment (GannonDetroit) [inline assignments/this.js:14]: why did you use a var?    try to stay in the habit of using let or const. But you're code satisfies the challenge/question asked.

File: assignments/this.js [hunk 8]
```diff
@@ -1,26 +1,53 @@
...
+console.log(x === window);
 // Principle 2
```
Comment (GannonDetroit) [inline assignments/this.js:16]: good job

File: assignments/this.js [hunk 9]
```diff
@@ -1,26 +1,53 @@
...
-
+const singer = {
+  name: "Queen",
+  favSong: "Mr Fahrenheit",
+  singIt: function() {
+    console.log(`${this.name} loves to sing ${this.favSong}`);
+  }
+};
+singer.singIt();
 // Principle 3
```
Comment (GannonDetroit) [inline assignments/this.js:26]: good job

File: assignments/this.js [hunk 10]
```diff
@@ -1,26 +1,53 @@
...
+  this.job = job;
+}
+
+const Mike = new Person({
+  name: "Mike",
+  age: 45,
+  job: "Butcher"
+});
+console.log(Mike);
 // Principle 4
```
Comment (GannonDetroit) [inline assignments/this.js:40]: good job

**Review by GannonDetroit — COMMENTED** [review]
Overall you did a great job.

I want to know why chunks of code are appearing to be copy and pasted or deleted and remade. Some extensions might be having an effect so let's look into that. 

Going forward I do NOT want to see only one commit. That's a huge problem for several reasons. So make sure to set reminders or do whatever it is you need to do to make sure this doesn't happen again. 

I'm glad you attempted the stretch on this. Well done.

**Comment by justman00** [comment]
Hey Gannon,

In terms of var, I probably did it out of instinct at first and forgot to edit after. But I did not delete anything actually or copy pasted any code, I have absolutely no idea why it does look like that😁

Von meinem iPhone gesendet

**Pull request merged**